packages/zowe-explorer/src/SpoolProvider.ts

```diff
@@ -84,9 +84,16 @@ export class SpoolFile {
 export function encodeJobFile(session: string, spool: zowe.IJobFile): vscode.Uri {
     ZoweLogger.trace("SpoolProvider.encodeJobFile called.");
     const query = JSON.stringify([session, spool]);
+
+    let path = `${spool.jobname}.${spool.jobid}.${spool.ddname}`;
+
+    if (spool.stepname) {
+        path = `${spool.jobname}.${spool.jobid}.${spool.stepname}.${spool.ddname}`;
+    }
+
```

Comment:
```ts
const spoolSegments = [
    spool.jobname,
    spool.jobid,
    spool.stepname,
    spool.ddname
];
const path = spoolSegments.filter((v) => v && v.length)).join(".");
```

Reply:
Good idea! I'll implement it.

I was wondering if it might be worth adding the procname to the path. I'll do some testing to see if the same duplicated name bug can occur if we're running different procs that have the same stepnames and the same DD names. if I can reproduce it, I'll add the procname to the array of segments as well